app/controllers/providers/means/identify_types_of_incomes_controller.rb

```diff
@@ -6,9 +6,20 @@ def show
       end
 
       def update
-        return continue_or_draft if none_selected? || transactions_added || draft_selected?
+        synchronize_credit_transaction_types
+
+        if none_selected?
+          legal_aid_application.update!(no_credit_transaction_types_selected: true)
+
+          return continue_or_draft
+        elsif transaction_types_selected?
+          legal_aid_application.update!(no_credit_transaction_types_selected: false)
+
+          return continue_or_draft
+        end
+
+        return continue_or_draft if draft_selected?
```

Comment:
This method doesn't feel particularly readable to me, and it feels quite long —especially for a controller action.

I think ideally it would be nice if the controller action looked something like this

```ruby
module Providers
  module Means
    class IdentifyTypesOfIncomesController < ProviderBaseController
      def update
        if income_types.save
          return continue_or_draft
        else
          render :show, status: :unprocessable_entity
        end
      end
      
      def income_types
        SomeWellNamedClass.new(...)
      end
    end
  end
end
```

with the complex logic refactored to a separate class.

It would simplify the request specs too, since the class would be covered by unit tests.

If this sounds worthwhile to you, let's schedule some time to pair on it!

Reply:
Lets discuss later but this was a refactor of existing, rather confusing, controller logic. It also changes the logic which was deleting all `transaction_types` and re-adding them using supplied params. This would not have been desirable with the cascade-delete logic.

Reply:
Agreed with @cpjmcquillan to make separate PR to refactor further.